From 6f2fc8b4bc920997c260e5e1a9f3ad520c9c815d Mon Sep 17 00:00:00 2001
From: nikhilad <nikhilad@ami.com>
Date: Fri, 7 Nov 2025 11:55:08 +0530
Subject: [PATCH] fansensor update regular expression to find pwm

Recreated the patch:

Update the regular expression to only find pwm* while excluding
pwm*_enable.

Signed-off-by: VinceChang6637 <vince_chang@aspeedtech.com>

Signed-off-by: nikhilad <nikhilad@ami.com>
---
 src/fan/FanMain.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fan/FanMain.cpp b/src/fan/FanMain.cpp
index 1563eb2..1cd7ed5 100644
--- a/src/fan/FanMain.cpp
+++ b/src/fan/FanMain.cpp
@@ -147,7 +147,7 @@ bool findPwmfanPath(unsigned int configPwmfanIndex,
 
     pwnfanDevName += std::to_string(configPwmfanIndex);
 
-    if (!findFiles(std::filesystem::path("/sys/class/hwmon"), R"(pwm\d+)",
+    if (!findFiles(std::filesystem::path("/sys/class/hwmon"), R"(pwm\d+$)",
                    pwmfanPaths))
     {
         lg2::error("No PWMs are found!");
-- 
2.34.1

